apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/SimpleManagedTracer.cs

```diff
@@ -292,5 +291,37 @@ private ImmutableStack<TraceSpan> TraceStack
             }
         }
 #endif
+
+        // TODO: Remove when https://github.com/dotnet/roslyn/issues/12255 is fixed and go back to using System.Collections.Immutable
+        private class ImmutableStack<T>
+        {
+            public static readonly ImmutableStack<T> Empty = new ImmutableStack<T>(default(T), null);
+
+            private ImmutableStack<T> _previous;
```

Comment:
Well what if we want to mutate the immutable stack at some point? Just kidding. Done.